refactor: [M3-6393] - MUI v5 - `Components > Snackbar`

jaalah-akamai opened the pull request:
## Description 📝
Refactors `Snackbar` to use `styled` api and cleaned up the overall usage of the component.
- [Reopened from previous PR](https://github.com/linode/manager/pull/9103)

## Major Changes 🔄
- Renamed `SnackBar` to `Snackbar` to follow naming convention
- We cannot update to latest version of `"notistack": "^3.0.1"` until some [issues](https://github.com/iamhosseindhv/notistack/pull/562) are addressed
- Converted `ToastNotifications` Storybook MDX to 7.0 CSF
- Using named export/imports
- Added new `info` colors to palette so we can have a true distinction between somethings that's informational vs success

## Pending Changes
- [x] Success color is now green instead of blue (pending UX review)

## Preview 📷
Should be no visual changes

![Screen Shot 2023-05-09 at 1 50 23 PM](https://github.com/linode/manager/assets/125309814/7c0efd75-cda0-4dc2-9b67-c570c2f47e8f)

## How to test 🧪
**Option 1:**
1. Run storybook `yarn storybook`
2. Go to http://localhost:6006/?path=/story/components-notifications-toasts--default

**Option 2:**
1. Run app locally and start a migration or resize on a linode (or anything that triggers a toast). See list here: https://github.com/linode/manager/blob/develop/packages/manager/src/features/ToastNotifications/ToastNotifications.tsx#L78
2. Verify toast messages still appear

bnussman-akamai commented on packages/manager/src/components/Snackbar/Snackbar.tsx:
```diff
@@ -29,7 +29,7 @@ const useStyles = makeStyles((theme: Theme) => ({
   },
   success: {
-    borderLeft: `6px solid ${theme.palette.primary.main}`,
+    borderLeft: `6px solid ${theme.color.green}`,
```
Instead of using `theme.color.green` can we fix `theme.palette.success.*` to actually be green and use it here?

```suggestion
    borderLeft: `6px solid ${theme.palette.success.main}`,
```

bnussman-akamai commented on packages/manager/src/features/Kubernetes/ClusterList/ClusterActionMenu.tsx:
```diff
@@ -19,15 +19,16 @@ interface Props {
 }
 
-type CombinedProps = Props & RouteComponentProps<{}> & WithSnackbarProps;
+type CombinedProps = Props & RouteComponentProps<{}>;
```
I know this is out of the scope of this ticket but we could fix this while where here... We don't need `RouteComponentProps`. We can remove `CombinedProps`, just use `Props`, and remove the `React.FunctionComponent`

```suggestion
```

bnussman-akamai commented on packages/manager/src/features/Kubernetes/ClusterList/ClusterActionMenu.tsx:
```diff
@@ -92,6 +93,6 @@ export const ClusterActionMenu: React.FunctionComponent<CombinedProps> = (
 };
 
-const enhanced = compose<CombinedProps, Props>(withSnackbar, withRouter);
+const enhanced = compose<CombinedProps, Props>(withRouter);
```
Related to other comment
```suggestion
```

jaalah-akamai commented:
@bnussman-akamai Good suggestions. I added a `info` palette since we're going to have one eventually. I updated notices that were clearly "informational" to stay with the blue. Success notifications and toasts will now be green! 🎉

bnussman-akamai reviewed (approved):
Looks good! Excited to have the 🟩. It feels way more fitting for success notices and toasts

jaalah-akamai commented:
@mjac0bs I'd use a `/` but the path is built from the title 
<img width="1047" alt="Screenshot 2023-07-05 at 9 29 32 PM" src="https://github.com/linode/manager/assets/125309814/2996179a-b86e-4ba7-b74d-913ff3a44d7f">
